### PR TITLE
Enable gocritic linter

### PR DESCRIPTION
**Pull request by ethanadams**
What: 
Add gocritic linter
Why:
Detect potential  bugs and issues.

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -109,7 +110,10 @@ linters-settings:
     simple: true
     range-loops: true # Report preallocation suggestions on range loops, true by default
     for-loops: false # Report preallocation suggestions on for loops, false by default
-
+  gocritic:
+    disabled-checks:
+    # Disabling appendAssign b/c it reports false positives
```
Comment (ifraixedes) [inline .golangci.yml]: Are there many false positives?

Comment (ethanadams) [reply .golangci.yml]: I reports 1 false positive.  

cmd/storj-sim/prefix.go:87:12: appendAssign: append result not assigned to the same slice (gocritic)
        buffer = append(writer.buffer, data...)
                 ^
make: *** [lint] Error 1

The assignment appears to be intentional b/c the assignment to write.buffer is deferred in the next line.

Comment (egonelbre) [reply .golangci.yml]: Note you can skip specific lines with an appropriate comment rather than disable the whole check.

Comment (ethanadams) [reply .golangci.yml]: Thanks for the tip.  I've add //nolint to the line and re-enabled the appendAssign check

File: cmd/storj-sim/prefix.go [hunk 2]
```diff
@@ -84,7 +84,7 @@ func (writer *prefixWriter) Write(data []byte) (int, error) {
 
 	// buffer everything that hasn't been written yet
 	if len(writer.buffer) > 0 {
-		buffer = append(writer.buffer, data...)
+		buffer = append(writer.buffer, data...) //nolint
```
Comment (ifraixedes) [inline cmd/storj-sim/prefix.go]: I would exclude only `gocritic` not all the linters

reference: https://github.com/golangci/golangci-lint#false-positives

File: cmd/uplink/cmd/share.go [hunk 3]
```diff
@@ -47,19 +47,20 @@ func init() {
 const shareISO8601 = "2006-01-02T15:04:05-0700"
 
 func parseHumanDate(date string, now time.Time) (*time.Time, error) {
-	if date == "" {
+	switch {
```
Comment (egonelbre) [inline cmd/uplink/cmd/share.go]: `switch date {`

File: cmd/storj-sim/prefix.go (hunk 2 above)
Comment (ethanadams) [reply cmd/storj-sim/prefix.go]: Good catch.  resolved an pushed

**Pull request merged**